protocols/common.go

```diff
@@ -0,0 +1,112 @@
+package protocols
```

Comment:
As a part of the tidy up (i.e. removing mention of state charts), I would have thought this file should be combined with `direct-funding.go`.